platform-sdk/swirlds-component-framework/src/main/java/com/swirlds/component/framework/schedulers/builders/internal/AbstractTaskSchedulerBuilder.java

```diff
@@ -264,11 +264,17 @@ public AbstractTaskSchedulerBuilder<OUT> withDataCounter(@NonNull ToLongFunction
      */
     @NonNull
     protected UncaughtExceptionHandler buildUncaughtExceptionHandler() {
-        if (uncaughtExceptionHandler != null) {
-            return uncaughtExceptionHandler;
-        } else {
-            return ExceptionHandlers.defaultExceptionHandler(name);
-        }
+        return uncaughtExceptionHandlerOr(ExceptionHandlers.defaultExceptionHandler(name));
+    }
+
+    /**
+     * Returns either the set uncaughtExceptionHandler or the handler sent as parameter.
+     * @param or the uncaught exception handler to return in case none was set
+     * @return the uncaught exception handler
+     */
+    @NonNull
+    protected UncaughtExceptionHandler uncaughtExceptionHandlerOr(@NonNull UncaughtExceptionHandler or) {
```

Comment:
I find this method and its name misleading. The method returns either the uncaughtExceptionHandler that was set on this builder or a provided default. That's unexpected behavior for a builder.